src/panels/lovelace/views/hui-sections-view.ts

```diff
@@ -0,0 +1,322 @@
+    if (!this._sectionConfigKeys.has(sectionConfig)) {
+      this._sectionConfigKeys.set(sectionConfig, Math.random().toString());
+    }
+    return this._sectionConfigKeys.get(sectionConfig)!;
+  }
+
+  protected render() {
+    if (!this.lovelace) return nothing;
+
+    const sectionsConfig = this._config?.sections ?? [];
+
+    const editMode = this.lovelace.editMode;
+
+    return html`
+      <ha-sortable
+        .disabled=${!editMode}
+        @item-moved=${this._sectionMoved}
+        group="section"
+        handle-selector=".handle"
+        draggable-selector=".section"
+        .rollback=${false}
+      >
+        <div class="container">
+          ${repeat(
+            sectionsConfig,
+            (sectionConfig) => this._getKey(sectionConfig),
+            (_sectionConfig, idx) => {
+              const section = this.sections[idx];
+              (section as any).itemPath = [idx];
+              return html`
+                <div class="section">
+                  ${editMode
+                    ? html`
+                        <div class="section-overlay">
+                          <div class="section-actions">
+                            <ha-svg-icon
+                              aria-hidden="true"
+                              class="handle"
+                              .path=${mdiArrowAll}
+                            ></ha-svg-icon>
+                            <ha-icon-button
+                              .label=${this.hass.localize("ui.common.edit")}
+                              @click=${this._editSection}
+                              .index=${idx}
+                              .path=${mdiPencil}
+                            ></ha-icon-button>
+                            <ha-icon-button
+                              .label=${this.hass.localize("ui.common.delete")}
+                              @click=${this._deleteSection}
+                              .index=${idx}
+                              .path=${mdiDelete}
+                            ></ha-icon-button>
+                          </div>
+                        </div>
+                      `
+                    : nothing}
+                  <div class="section-wrapper">${section}</div>
+                </div>
+              `;
+            }
+          )}
+          ${editMode
+            ? html`
+                <button
+                  class="add"
+                  @click=${this._addSection}
+                  aria-label=${this.hass.localize(
+                    "ui.panel.lovelace.editor.section.add_section"
+                  )}
+                  .title=${this.hass.localize(
+                    "ui.panel.lovelace.editor.section.add_section"
+                  )}
+                >
+                  <ha-svg-icon .path=${mdiViewGridPlus}></ha-svg-icon>
+                </button>
+              `
+            : nothing}
+        </div>
+      </ha-sortable>
+    `;
+  }
+
+  private _addSection(): void {
```

Comment:
I would love to move the editor code out of here in a future PR, so it will not slow down none edit usage

Reply:
Me too! I had some issue with sortable when trying to move it out but it should be solvable!